tests/Jellyfin.Providers.Tests/MediaInfo/AudioResolverTests.cs

```diff
@@ -0,0 +1,177 @@
+
+        public AudioResolverTests()
+        {
+            var englishCultureDto = new CultureDto
+            {
+                Name = "English",
+                DisplayName = "English",
+                ThreeLetterISOLanguageNames = new[] { "eng" },
+                TwoLetterISOLanguageName = "en"
+            };
+
+            var localizationManager = new Mock<ILocalizationManager>(MockBehavior.Loose);
+            localizationManager.Setup(lm => lm.FindLanguageInfo(It.IsRegex(@"en.*", RegexOptions.IgnoreCase)))
+                .Returns(englishCultureDto);
+
+            var mediaEncoder = new Mock<IMediaEncoder>(MockBehavior.Strict);
+            mediaEncoder.Setup(me => me.GetMediaInfo(It.IsAny<MediaInfoRequest>(), It.IsAny<CancellationToken>()))
+                .Returns<MediaInfoRequest, CancellationToken>((_, _) => Task.FromResult(new MediaBrowser.Model.MediaInfo.MediaInfo
+                {
+                    MediaStreams = new List<MediaStream>
+                    {
+                        new()
+                    }
+                }));
+
+            _audioResolver = new AudioResolver(localizationManager.Object, mediaEncoder.Object, new NamingOptions());
+        }
+
+        [Fact]
+        public async void AddExternalAudioStreams_GivenMixedFilenames_ReturnsValidSubtitles()
+        {
+            var startIndex = 0;
+            var index = startIndex;
+            var files = new[]
+            {
+                DirectoryPath + "/My.Video.mp3",
+                DirectoryPath + "/Some.Other.Video.mp3",
+                DirectoryPath + "/My.Video.png",
+                DirectoryPath + "/My.Video.srt",
+                DirectoryPath + "/My.Video.txt",
+                DirectoryPath + "/My.Video.vtt",
+                DirectoryPath + "/My.Video.ass",
+                DirectoryPath + "/My.Video.sub",
+                DirectoryPath + "/My.Video.ssa",
+                DirectoryPath + "/My.Video.smi",
+                DirectoryPath + "/My.Video.sami",
+                DirectoryPath + "/My.Video.en.mp3",
+                DirectoryPath + "/My.Video.Label.mp3",
+                DirectoryPath + "/My.Video.With Additional Garbage.en.aac",
+                DirectoryPath + "/My.Video.With Additional Garbage.en.aac",
+                DirectoryPath + "/My.Video.With.Additional.Garbage.en.mp3"
+            };
+            var expectedResult = new[]
+            {
+                CreateMediaStream(DirectoryPath + "/My.Video.mp3", null, null, index++),
+                CreateMediaStream(DirectoryPath + "/My.Video.en.mp3", "eng", null, index++),
+                CreateMediaStream(DirectoryPath + "/My.Video.Label.mp3", null, "Label", index++),
+                CreateMediaStream(DirectoryPath + "/My.Video.With Additional Garbage.en.aac", "eng", "With Additional Garbage", index++),
+                CreateMediaStream(DirectoryPath + "/My.Video.With Additional Garbage.en.aac", "eng", "With Additional Garbage", index++),
```

Comment:
Remove duplicated path

```suggestion
                DirectoryPath + "/My.Video.With Additional Garbage.en.aac",
                DirectoryPath + "/My.Video.With.Additional.Garbage.en.mp3"
            };
            var expectedResult = new[]
            {
                CreateMediaStream(DirectoryPath + "/My.Video.mp3", null, null, index++),
                CreateMediaStream(DirectoryPath + "/My.Video.en.mp3", "eng", null, index++),
                CreateMediaStream(DirectoryPath + "/My.Video.Label.mp3", null, "Label", index++),
                CreateMediaStream(DirectoryPath + "/My.Video.With Additional Garbage.en.aac", "eng", "With Additional Garbage", index++),
```